blobstore/src/test/java/org/jclouds/blobstore/strategy/internal/PrefixTest.java

```diff
@@ -0,0 +1,116 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.blobstore.strategy.internal;
+
+import static org.testng.Assert.assertEquals;
+
+import java.io.File;
+
+import org.jclouds.ContextBuilder;
+import org.jclouds.blobstore.BlobStore;
+import org.jclouds.blobstore.domain.StorageMetadata;
+import org.jclouds.blobstore.domain.StorageType;
+import org.jclouds.blobstore.options.ListContainerOptions;
+import org.jclouds.util.Closeables2;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.google.common.collect.Iterables;
+import com.google.inject.Injector;
+
+@Test(testName = "PrefixTest", singleThreaded = true)
+public class PrefixTest {
+   private BlobStore blobStore;
+   private ConcatenateContainerLists concatter;
+
+   @BeforeClass
+   public void setupBlobStore() {
+      Injector injector = ContextBuilder.newBuilder("transient").buildInjector();
+      blobStore = injector.getInstance(BlobStore.class);
+      concatter = injector.getInstance(ConcatenateContainerLists.class);
+   }
+
+   @AfterClass
+   public void closeBlobSore() {
+      if (blobStore != null) {
+         Closeables2.closeQuietly(blobStore.getContext());
+      }
+   }
+
+   public void testListWithPrefix() {
+      String containerName = "prefix";
+      String prefix = "foo";
+      blobStore.createContainerInLocation(null, containerName);
+      blobStore.putBlob(containerName, blobStore.blobBuilder(prefix).payload("").build());
+      blobStore.putBlob(containerName, blobStore.blobBuilder(prefix + "bar").payload("").build());
+      blobStore.putBlob(containerName, blobStore.blobBuilder(prefix + "baz").payload("").build());
+      blobStore.putBlob(containerName, blobStore.blobBuilder("bar").payload("").build());
+
+      Iterable<? extends StorageMetadata> results = concatter.execute(containerName,
+            ListContainerOptions.Builder.prefix(prefix));
+      assertEquals(Iterables.size(results), 3);
+      assertEquals(Iterables.get(results, 0).getName(), prefix);
+      assertEquals(Iterables.get(results, 0).getType(), StorageType.BLOB);
+      assertEquals(Iterables.get(results, 1).getName(), prefix + "bar");
+      assertEquals(Iterables.get(results, 1).getType(), StorageType.BLOB);
+      assertEquals(Iterables.get(results, 2).getName(), prefix + "baz");
+      assertEquals(Iterables.get(results, 2).getType(), StorageType.BLOB);
+   }
+
+   public void testListWithPrefixAndSeparator() {
+      String containerName = "prefixWithSeparator";
+      String prefix = "foo";
+      blobStore.createContainerInLocation(null, containerName);
+      blobStore.putBlob(containerName, blobStore.blobBuilder(prefix + File.separator + "object").payload("").build());
```

Comment:
Do you really want to use `File.separator` since this will differ when running on Linux vs. Windows.